lib/index.js

```diff
@@ -1,55 +1,65 @@
 var fs    = require('fs');
 var Joi   = require('joi');
+var glob  = require('glob');
 var path  = require('path');
 
+var internals = {};
+
 exports.register = function (server, options, next) {
 
+  internals.bookshelf = null;
+  internals.modelObjects = [];
+
   var schema = {
     knex: Joi.object().required(),
-    plugins: Joi.array().includes(Joi.string()).default([]),
-    models: Joi.string().required(),
+    plugins: Joi.array().items(Joi.string()).default([]),
+    models: Joi.array().items(Joi.string()).single().required(),
     base: Joi.func().optional(),
     namespace: Joi.string().optional()
   };
 
-  try {
-    Joi.assert(options, schema, 'Invalid Configuration Object');
-  } catch (ex) {
-    return next(ex);
+  var result = Joi.validate(options, schema);
+  if (result.error) {
+    next(result.error);
   }
+  options = result.value;
 
-  var bookshelf;
   try {
-    bookshelf = require('bookshelf')(require('knex')(options.knex));
+    internals.bookshelf = require('bookshelf')
+        (require('knex')(options.knex));
   } catch (ex) {
     return next(new Error('Bad Knex Options: ' + ex.toString()));
   }
 
   options.plugins.map(function (plugin) {
-    bookshelf.plugin(plugin);
+    internals.bookshelf.plugin(plugin);
   });
 
-  var baseModel;
   if (options.base) {
-    baseModel = options.base(bookshelf);
+    internals.baseModel = options.base(internals.bookshelf);
   } else {
-    baseModel = bookshelf.Model.extend({});
+    internals.baseModel = internals.bookshelf.Model.extend({});
   }
 
-  fs.readdirSync(options.models).forEach(function (model) {
-    var modelDir = path.resolve(options.models);
-    var modelName = path.basename(model).replace(path.extname(model), '');
-    if (!/^\..*/.test(modelName)) {
-      modelName = modelName[0].toUpperCase() + modelName.substr(1);
-      bookshelf.model(modelName,
-        require(path.join(modelDir, model))(baseModel, bookshelf));
+  var gFiles = [];
+  for (var i = options.models.length - 1; i >= 0; i--) {
+    var filename = options.models[i];
+    if (!glob.hasMagic(filename)) {
+      var stats = fs.lstatSync(filename);
+      if (stats.isDirectory()) {
+        filename = path.join(filename, '*.js');
+      }
     }
-  });
+    gFiles = glob.sync(filename);
+    internals.modelObjects = internals.modelObjects.concat(gFiles);
+  }
+
+  internals._loadModels();
```

Comment:
Since `_loadModels` is only called once now, I don't think it needs to be extracted out to a function anymore.